pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -304,24 +304,36 @@ func (r *ReconcileApplicationLayer) Reconcile(ctx context.Context, request recon
 
 // updateApplicationLayerWithDefaults populates the applicationlayer with defaults.
 func updateApplicationLayerWithDefaults(al *operatorv1.ApplicationLayer) {
-	defaultLogIntervalSeconds := int64(5)
-	defaultLogRequestsPerInterval := int64(-1)
+	var defaultLogIntervalSeconds int64 = 5
+	var defaultLogRequestsPerInterval int64 = -1
+	var defaultLogCollectionDisabled operatorv1.LogCollectionStatusType = operatorv1.L7LogCollectionDisabled
+	var defaultWebApplicationFirewallDisabled operatorv1.WAFStatusType = operatorv1.WAFDisabled
 
-	if al.Spec.LogCollection != nil {
+	if al.Spec.LogCollection == nil {
+		al.Spec.LogCollection = new(operatorv1.LogCollectionSpec)
+		al.Spec.LogCollection.CollectLogs = &defaultLogCollectionDisabled
+	}
+
+	if *al.Spec.LogCollection.CollectLogs == operatorv1.L7LogCollectionEnabled {
```

Comment:
We're de-referencing a pointer that might be nil here. Imaging the case where a user provides this:

```go
LocCollection{
  LogRequestsPerInterval: 5
}
```

The struct is non-nil, so we won't enter the branch above, but the `CollectLogs` field is nil, so we'll panic here.

Reply:
I think the fix is to remove seeing the default in the branch above, and instead do something like this:


```go

// Determine if any fields were specified. If any fields were provided, that signals intent by the user 
// to enable log collection.
shouldEnableLogCollection := collectLogs != nil || logInterval != nil || logIntervalSeconds != nil
if al.Spec.LogCollection.CollectLogs == nil {
  if shouldEnableLogCollection {
    al.Spec.LogCollection.CollectLogs = Enabled
  } else {
      al.Spec.LogCollection.CollectLogs = Disabled
  }
}

if al.Spec.LogCollection.CollectLogs == Enabled {
  // default other log collection fields
}
```

Reply:
Yup, good point. I think this is resolved now, I didn't consider the structure as provided.

The default value for `CollectLogs` is `Disabled` though so I think in this example case L7 logs should remain disabled. I can change that if you want this feature indirectly enabled by other parameters - let me know.

Reply:
If the user provided intent to enable log collection (by setting one of the other log collection fields) but didn't explicitly set the collectLogs field and left it nil, we have two options:

- Default the CollectLogs field unconditionally to "Disabled", resulting in an invalid configuration and returning an error when we reach the validation step.
- Default the CollectLogs field conditionally in a way that is consistent with the other configuration that is provided (Disabled if no other fields set, Enabled if other fields are set). 

The latter is what I would recommend. If we want to go with the former, we should probably be marking the CollectLogs field as required in the API to make sure any request is rejected if it includes the LogCollection substruct without the CollectLogs field being set.

I can see a case for either approach, but the latter feels more "useful" to me and less frustrating to the user.

Reply:
That's a new behavioral change though so if you don't want to deal with it in this PR that's OK.